src/schemas.jl

```diff
@@ -254,35 +254,58 @@ accepted_field_type(::SchemaVersion, ::Type{Symbol}) = Union{Symbol,String}
+schema violations, see [`Legolas.find_violations`](@ref).
 
-- Define `A = Legolas.accepted_field_type(sv, F)`
-- If `f::T` is present in `ts`, ensure that `T <: A` or else immediately return `f::Symbol => T::DataType`.
-- If `f` isn't present in `ts`, ensure that `Missing <: A` or else immediately return `f::Symbol => missing::Missing`.
-
-Otherwise, return `nothing`.
-
-See also: [`Legolas.validate`](@ref), [`Legolas.complies_with`](@ref)
+See also: [`Legolas.find_violations`](@ref), [`Legolas.validate`](@ref), [`Legolas.complies_with`](@ref)
 """
 find_violation(::Tables.Schema, sv::SchemaVersion) = throw(UnknownSchemaVersionError(sv))
 
-function _find_violation end
+"""
+    Legolas.find_violations(ts::Tables.Schema, sv::Legolas.SchemaVersion)
+
+Return vector of all schema violations for table `ts` and schema `sv`; otherwise,
+return `Any[]`.
+
+A schema violation occurs when a required field `f` of `sv` is not present in `ts`,
+reported as `f::Symbol => missing::Missing`, or when a required field `f` does not match
+the expected type `T`, reported as `f::Symbol => T::DataType`.
+
+See also: [`Legolas.find_violation`](@ref), [`Legolas.validate`](@ref), [`Legolas.complies_with`](@ref)
+"""
+find_violations(::Tables.Schema, sv::SchemaVersion) = throw(UnknownSchemaVersionError(sv))
 
 """
     Legolas.validate(ts::Tables.Schema, sv::Legolas.SchemaVersion)
 
-Throws a descriptive `ArgumentError` if `!isnothing(find_violation(ts, sv))`,
-otherwise return `nothing`.
+Throws a descriptive `ArgumentError` if any violations are found.
 
-See also: [`Legolas.find_violation`](@ref), [`Legolas.complies_with`](@ref)
+See also: [`Legolas.find_violations`](@ref), [`Legolas.find_violation`](@ref), [`Legolas.complies_with`](@ref)
 """
 function validate(ts::Tables.Schema, sv::SchemaVersion)
-    result = find_violation(ts, sv)
-    isnothing(result) && return nothing
-    field, violation = result
-    ismissing(violation) && throw(ArgumentError("could not find expected field `$field` in $ts"))
-    expected = getfield(required_fields(sv), field)
-    throw(ArgumentError("field `$field` has unexpected type; expected <:$expected, found $violation"))
+    results = find_violations(ts, sv)
+    isempty(results) && return nothing
+
+    field_err = Symbol[]
+    type_err = Tuple{Symbol,Type,Type}[]
+    for result in results
+        field, violation = result
+        if ismissing(violation)
+            push!(field_err, field)
+        else
+            expected = getfield(required_fields(sv), field)
+            push!(type_err, (field, expected, violation))
+        end
+    end
+    err_msg = "Tables.Schema violates Legolas schema `$(string(name(sv), "@", version(sv)))`:\n"
+    for err in field_err
+        err_msg *= "  - Could not find required field: `$err`\n"
+    end
+    for (field, expected, violation) in type_err
+        err_msg *= "  - Incorrect type: `$field` expected `<:$expected`, found `$violation`\n"
+    end
```

Comment:
We should probably mirror the indentation level used by `Table.Schema`
```suggestion
    for err in field_err
        err_msg *= " - Could not find required field: `$err`\n"
    end
    for (field, expected, violation) in type_err
        err_msg *= " - Incorrect type: `$field` expected `<:$expected`, found `$violation`\n"
    end
```